app/features/user/user-repository.ts

```diff
@@ -83,6 +84,10 @@ export class UserRepository {
     const { data: updatedUser, error } = await query.single();
 
     if (error) {
+      if (error.code === '23505') {
+        throw new UniqueConstraintError();
```

Comment:
lets pass error message inside so we get the message about what violated the constraint too